src/services/services.ts

```diff
@@ -2088,10 +2088,15 @@ namespace ts {
             const token = findPrecedingToken(position, sourceFile);
             if (!token) return undefined;
             const element = token.kind === SyntaxKind.GreaterThanToken && isJsxOpeningElement(token.parent) ? token.parent.parent
-                : isJsxText(token) ? token.parent : undefined;
+                : isJsxText(token) && isJsxElement(token.parent) ? token.parent : undefined;

```

Comment:
To comment on why this is necessary, I found that `JsxText.parent` is not necessarily `JsxElement` but it can be `JsxFragment`, even though typing-wise it's supposed to be `JsxElement` according to https://github.com/microsoft/TypeScript/blob/8362a0f929d74ff46828016ec67c05744a8dbb3c/src/compiler/types.ts#L2601-L2605

Without this change, `isUnclosedTag` will lead to runtime error since `openingElement` becomes `undefined`, for example, in the test case `const x = <> foo/*1*/ </>` (see `@Filename: /1.tsx` in `autoCloseFragment.ts`).

Probably, a legitimate fix would be something like defining `type JsxParent = JsxElement | JsxFragment` and replace existing member types `parent: JsxElement` with `parent: JsxParant`, but I'm not so sure yet that is what's supposed to be since I'm very new to the code base.
I would appreciate if someone familar with these parts can comment on this.

Reply:
As far as I can tell, I think this is a good way to fix this problem. I think maybe the original code just didn't consider `JsxFragments` when it was written.

Reply:
@hi-ogawa did you try changing the type of `parent` here to be `JsxElement | JsxFragment`? I agree we should really fix that type. Would you mind trying it and seeing if you can resolve any other errors that come up?

Reply:
Thanks for the feedback!
Well, I didn't try anything yet, but I'll see what I can do and get back to you with my findings.

Reply:
@andrewbranch Following your request, I addressed the issue on this commit https://github.com/microsoft/TypeScript/pull/45532/commits/19cc4a3ef5bab28bd087017b72a14d5a517372b0.
There wasn't any type error related to `JsxText.parent` so it looks alright.
I also changed the type of `JsxExpression.parent` to include `JsxFragment` since I believe that's what it should be, then there was one type error in `src/services/completions.ts`, so I did a trivial fix for that.
Overall, I think there aren't much use `JsxText.parent` or `JsxExpression.parent` in the code base currently, so hopefully the type change won't have much impact.
But, one thing I worry about is that it's going to change public API (as you can see from baselines/references/api), so the library users will be also affected by this change.